package filesync;
message SyncInfo
{//告知哪些文件有修改
    required int32 id = 1;    //0是创建文件夹，1是create文件，2是modify,3是删除，4是重命名
    required string filename = 2;   //文件名（包含相对顶层文件夹的路径）
    optional string newFilename = 3;    //新文件名(用于重命名)
    optional int32 size = 4;  //文件大小
    optional string MD5 = 5;    //文件的md5
}

message SignIn
{//登录
    required string username = 1;
    required string password = 2;
}

message Init
{//初始化
    repeated SyncInfo info = 3;
}

message SendFile
{//告知哪些文件需要发送或接受
    required int32 id = 1;  //消息标识，0:服务器传给客户端,1:客户端传给服务器
    required string filename = 2;   
}

//服务端没有，客户端有的【空洞】
message HoleInfo
{
    required int32 start = 1;
    required int32 length = 2;
}

message Sendfile_Rsync
{
    repeated HoleInfo;
}

message IsControl
{//这是传输通道
    required int32 id = 1;
}

//客户端向服务端发送rsync所需的各块的adler32和md5
message BlockInfo
{
    required int32 adler32 = 1;
    required int32 md5 = 2;
    required int32 offset = 3;    //此数据块在原文件中的位置
}

message RsyncInfo
{
    required string filename = 1;
    required repeated BlockInfo = 2;
}
/*****以上是控制通道负责*********/
/*****以下是文件传输通道********/
message FileInfo
{
    required int32 size = 1;    //文件大小
    required string filename = 2;//文件名
}




